controllers/uninstall_test.go

```diff
@@ -0,0 +1,168 @@
+				Name: "cluster",
+			},
+		})).To(Succeed())
+		Expect(k8sClient.Delete(context.Background(), &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "testsecret",
+				Namespace: DestinationNamespace,
+			},
+		})).To(Succeed())
+	})
+
+	BeforeEach(func() {
+		// Create openshift-monitoring namespace because metrics stands up prometheus endpoint here
+		err := k8sClient.Create(context.Background(), &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "openshift-monitoring",
+			},
+			Spec: corev1.NamespaceSpec{},
+		})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			Expect(err).To(BeNil())
+		}
+
+		// Create target namespace
+		err = k8sClient.Create(context.Background(), &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: DestinationNamespace,
+			},
+			Spec: corev1.NamespaceSpec{},
+		})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			Expect(err).To(BeNil())
+		}
+		// Create ClusterVersion
+		// Attempted to Store Version in status. Unable to get it to stick.
+		Expect(k8sClient.Create(context.Background(), &configv1.ClusterVersion{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "version",
+			},
+			Spec: configv1.ClusterVersionSpec{
+				Channel:   "stable-4.9",
+				ClusterID: "12345678910",
+			},
+		})).To(Succeed())
+
+		// Create ClusterIngress
+		// Attempted to Store Version in status. Unable to get it to stick.
+		Expect(k8sClient.Create(context.Background(), &configv1.Ingress{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cluster",
+			},
+			Spec: configv1.IngressSpec{
+				Domain: "apps.installer-test-cluster.dev00.red-chesterfield.com",
+			},
+		})).To(Succeed())
+
+		Expect(k8sClient.Create(context.Background(), &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "hypershift-deployment",
+				Namespace: DestinationNamespace,
+			},
+		})).To(Succeed())
+
+		// Create test secret in target namespace
+		testsecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "testsecret",
+				Namespace: DestinationNamespace,
+			},
+		}
+		Eventually(func() error {
+			err := k8sClient.Create(context.TODO(), testsecret)
+			if apierrors.IsAlreadyExists(err) {
+				return nil
+			}
+			return err
+		}, timeout, interval).Should(Succeed())
+
+	})
+
+	When("creating a new BackplaneConfig", func() {
+
+		Context("and no image pull policy is specified", func() {
+			It("should remove old components", func() {
+				// createCtx := context.Background()
+				By("creating the backplane config")
+				backplaneConfig := &v1.MultiClusterEngine{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "multicluster.openshift.io/v1",
+						Kind:       "MultiClusterEngine",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: BackplaneConfigName,
+					},
+					Spec: v1.MultiClusterEngineSpec{
+						TargetNamespace: DestinationNamespace,
+						ImagePullSecret: "testsecret",
+					},
+				}
+
+				reconciler.ensureRemovalsGone(backplaneConfig)
```

Comment:
Why does this function need to be called directly? Shouldn't it just happen as part of the normal reconcile loop?